### PR TITLE
Feature/build-script

### PR DESCRIPTION
**Pull request by Kas-tle**
This PR:
- Uses awk to one-line all JSON files in the resource pack
- Adds comments to document all parts of the build script
- Updates the readme regarding build-script use and current implements
- Renames the buildscript to `prepare_pack.sh` to denote that it is no longer only used for file copying

### CODE REVIEW

**Comment by lukeeey** [comment]
If you're going to minify the JSON, I recommend including some sort of file in the finished resource pack that provides some information and links back to this repo, in case anyone stumbles upon it and would like to take a look.

Also, you should strip comments when minifying. Although there are none now if some unsuspecting person were to modify the source from this repo and add comments then try and minify it for production then I'm not quite sure that would work..

**Comment by Camotoy** [comment]
We can add a link in the README, which we don't strip out from the final product.

**Comment by Kas-tle** [comment]
@lukeeey 

I'll add a note to the readme to address the first point.

As for stripping comments... I would say just don't add comments in general. While you're correct that Minecraft does parse them, it's generally an annoyance since most JSON editors will show them as an error, since they aren't part of the JSON standard. Also, if we ever decide to lint the pack, most JSON linters just will not support comments.

File: prepare_pack.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Create required sprites with Imagemagick
+convert -append extracted/assets/minecraft/textures/particle/sweep_*.png -define png:format=png8 textures/particle/sweep_attack.png
+
+rm client.jar
+rm -r extracted
+
+# Set our input field seperator back to new line
+IFS=$'\n'
+
+# One-line our JSON files if the script is called with -jc (./copy_java_files_to_pack.sh -jc)
```
Comment (Camotoy) [inline prepare_pack.sh:25]: This comment needs to be updated.

**Comment by lukeeey** [comment]
> @lukeeey
> 
> I'll add a note to the readme to address the first point.
> 
> As for stripping comments... I would say just don't add comments in general. While you're correct that Minecraft does parse them, it's generally an annoyance since most JSON editors will show them as an error, since they aren't part of the JSON standard. Also, if we ever decide to lint the pack, most JSON linters just will not support comments. I think adding what we do to developer_documentation.md is a better way of going about this.

I agree with all of this, but the sad fact is that Mojang supports comments for some reason and (correct me if I’m wrong) uses comments in their own resource pack JSON files, which means if anyone even copies a file from the vanilla resource pack then it won’t minify correctly.

If you can’t implement it easily or don’t want to then i won’t make you, it’s not the biggest problem.

**Comment by Kas-tle** [comment]
> I agree with all of this, but the sad fact is that Mojang supports comments for some reason and (correct me if I’m wrong) uses comments in their own resource pack JSON files, which means if anyone even copies a file from the vanilla resource pack then it won’t minify correctly.
> 
> If you can’t implement it easily or don’t want to then i won’t make you, it’s not the biggest problem.

It might be possible with a second awk expression... though it just makes it somewhat complex within the limits of what you can do with awk, since you then also have to make sure the comment isn't actually in quotes or something and therefore not a comment. We were trying to avoid installing anything extra on the CI, but if you can think of a way to accomplish it with awk I would certainly put it in in case someone tries to merge this with another pack. I'll see if I can figure it out.

File: prepare_pack.sh (hunk 1 above)
Comment (Kas-tle) [reply prepare_pack.sh]: Resolved by 4b964a0.

**Comment by Camotoy** [comment]
I'll merge as-is and we can address this later if need be.

**Pull request merged**